### PR TITLE
Lots of changes to the server configuration pages

### PR DESCRIPTION
**Pull request by dkanada**
I have been trying to simplify the layout of the server settings. I will list some of the larger changes below in no particular order. It could use some testing once I move a few more things around.

1. The plugin catalog was cloned in five different sections including the old sync interface. This was in the form of `appservices.js` which was a feature specific plugin page, which I removed in favor of the main plugin page.
2. There was a network share path setting inside the directory browser component which seemed like an odd place for a setting. I moved it to the General section.
3. The DLNA page had dead code referencing the streaming bitrate limit, which is currently in the playback section.
4. I was thinking of moving all path related settings into the General tab and linking it from the Paths widget on the dashboard. (left the transcode and ffmpeg paths in the Transcoding section for now, wanted to get feedback before moving those two)

### CODE REVIEW

File: src/dashboard/dashboardgeneral.js [hunk 1]
```diff
@@ -2,50 +2,118 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox", "emby-textarea", "emb
...
-            var promise1 = ApiClient.getServerConfiguration(),
-                promise2 = ApiClient.getJSON(ApiClient.getUrl("Localization/Options")),
-                promise3 = ApiClient.getSystemInfo();
-            Promise.all([promise1, promise2, promise3]).then(function(responses) {
-                loadPage(view, responses[0], responses[1], responses[2])
-            }), ApiClient.getNamedConfiguration(brandingConfigKey).then(function(config) {
-                currentBrandingOptions = config, view.querySelector("#txtLoginDisclaimer").value = config.LoginDisclaimer || "", view.querySelector("#txtCustomCss").value = config.CustomCss || ""
+        });
+
+        $("#btnSelectMetadataPath", view).on("click.selectDirectory", function() {
```
Comment (JustAMan) [inline src/dashboard/dashboardgeneral.js]: these two functions seem identical except what field they modify and what help text show, can copy-pasteness be reduced here?

File: src/dashboard/librarysettings.js [hunk 2]
```diff
@@ -2,45 +2,47 @@ define(["jQuery", "loading", "libraryMenu", "fnchecked", "emby-checkbox", "emby-
     "use strict";
 
     function loadPage(page, config) {
-        config.MergeMetadataAndImagesByName ? $(".fldImagesByName", page).hide() : $(".fldImagesByName", page).show(), $("#chkSaveMetadataHidden", page).checked(config.SaveMetadataHidden), $("#txtMetadataPath", page).val(config.MetadataPath || ""), $("#txtMetadataNetworkPath", page).val(config.MetadataNetworkPath || ""), loading.hide()
```
Comment (JustAMan) [inline src/dashboard/librarysettings.js]: why check for `config.MergeMetadataAndImagesByName` is gone?

File: src/dashboard/librarysettings.js [hunk 3]
```diff
@@ -2,45 +2,47 @@ define(["jQuery", "loading", "libraryMenu", "fnchecked", "emby-checkbox", "emby-
...
-        loading.hide(), response && 404 === response.status ? alertText("The metadata path entered could not be found. Please ensure the path is valid and try again.") : response && 500 === response.status && alertText("The metadata path entered is not valid. Please ensure the path exists and that Jellyfin server has write access to the folder.")
+            alert(options);
+        });
     }
 
     function onSubmit() {
         loading.show();
         var form = this;
-        return ApiClient.getServerConfiguration().then(function(config) {
-            config.SaveMetadataHidden = $("#chkSaveMetadataHidden", form).checked(), config.EnableTvDbUpdates = $("#chkEnableTvdbUpdates", form).checked(), config.EnableTmdbUpdates = $("#chkEnableTmdbUpdates", form).checked(), config.EnableFanArtUpdates = $("#chkEnableFanartUpdates", form).checked(), config.MetadataPath = $("#txtMetadataPath", form).val(), config.MetadataNetworkPath = $("#txtMetadataNetworkPath", form).val(), config.FanartApiKey = $("#txtFanartApiKey", form).val(), ApiClient.updateServerConfiguration(config).then(Dashboard.processServerConfigurationUpdateResult, onSubmitFail)
```
Comment (JustAMan) [inline src/dashboard/librarysettings.js]: why did you remove `EnableTvDbUpdates`, `chkEnableTmdbUpdates` and `chkEnableFanartUpdates` checks here?

File: src/dashboard/librarysettings.js [hunk 4]
```diff
@@ -61,34 +63,25 @@ define(["jQuery", "loading", "libraryMenu", "fnchecked", "emby-checkbox", "emby-
...
+            ApiClient.getNamedConfiguration("metadata").then(function(metadata) {
                 loadMetadataConfig(page, metadata)
-            }), ApiClient.getNamedConfiguration("fanart").then(function(metadata) {
+            });
+            ApiClient.getNamedConfiguration("fanart").then(function(metadata) {
                 loadFanartConfig(page, metadata)
-            }), ApiClient.getSystemInfo().then(function(info) {
+            });
+            ApiClient.getSystemInfo().then(function(info) {
                 "Windows" === info.OperatingSystem ? page.querySelector(".fldSaveMetadataHidden").classList.remove("hide") : page.querySelector(".fldSaveMetadataHidden").classList.add("hide")
```
Comment (JustAMan) [inline src/dashboard/librarysettings.js]:
```suggestion
                 if ("Windows" === info.OperatingSystem) {
                     page.querySelector(".fldSaveMetadataHidden").classList.remove("hide");
                 } else {
                     page.querySelector(".fldSaveMetadataHidden").classList.add("hide");
                 }
```

File: src/dashboard/livetvtuner.js [hunk 5]
```diff
@@ -2,31 +2,54 @@ define(["globalize", "loading", "libraryMenu", "dom", "emby-input", "emby-button
...
     function isM3uVariant(type) {
-        return -1 !== ["nextpvr"].indexOf(type || "")
+        return ["nextpvr"].indexOf(type || "") !== -1;
     }
 
     function fillTypes(view, currentId) {
-        return ApiClient.getJSON(ApiClient.getUrl("LiveTv/TunerHosts/Types")).then(function(types) {
+        ApiClient.getJSON(ApiClient.getUrl("LiveTv/TunerHosts/Types")).then(function(types) {
             var selectType = view.querySelector(".selectType");
             selectType.innerHTML = types.map(function(t) {
```
Comment (JustAMan) [inline src/dashboard/livetvtuner.js]: `t` -> `tunerType` ?

Comment (JustAMan) [inline src/dashboard/livetvtuner.js]: I would rather we build `innerHTML` as a separate variable and assign it once... I saw issues (on some really old browser) when select element was built incrementally, plus it makes no sense to rebuild DOM all the time instead of rebuilding it once.

File: src/dashboard/livetvtuner.js [hunk 6]
```diff
@@ -2,31 +2,54 @@ define(["globalize", "loading", "libraryMenu", "dom", "emby-input", "emby-button
...
 
     function fillTunerHostInfo(view, info) {
-        var selectType = view.querySelector(".selectType"),
-            type = info.Type || "";
-        info.Source && isM3uVariant(info.Source) && (type = info.Source), selectType.value = type, onTypeChange.call(selectType), view.querySelector(".txtDevicePath").value = info.Url || "", view.querySelector(".txtFriendlyName").value = info.FriendlyName || "", view.querySelector(".txtUserAgent").value = info.UserAgent || "", view.querySelector(".fldDeviceId").value = info.DeviceId || "", view.querySelector(".chkFavorite").checked = info.ImportFavoritesOnly, view.querySelector(".chkTranscode").checked = info.AllowHWTranscoding, view.querySelector(".chkStreamLoop").checked = info.EnableStreamLooping, view.querySelector(".txtTunerCount").value = info.TunerCount || "0"
+        var selectType = view.querySelector(".selectType");
+        var type = info.Type || "";
+        if (info.Source && isM3uVariant(info.Source)) {
+            selectType.value = info.Source;
+            onTypeChange.call(selectType);
```
Comment (JustAMan) [inline src/dashboard/livetvtuner.js]: This change looks wrong, you're missing a case when `info.Source && isM3uVariant(...)` is false but `info.Type` is non-empty. Previously it was assigned after this check, now it's lost.

Correct code looks like this IMHO:
```js
var type = info.Type || "";
if (info.Source && isM3uVariant(info.Source)) {
    type = info.Source;
}
selectType.value = type;
onTypeChange.call(selectType);
```

File: src/dashboard/livetvtuner.js [hunk 7]
```diff
@@ -74,16 +97,6 @@ define(["globalize", "loading", "libraryMenu", "dom", "emby-input", "emby-button
         })
     }
 
-    function getTabs() {
```
Comment (JustAMan) [inline src/dashboard/livetvtuner.js]: why is this check gone?

File: src/dashboardgeneral.html [hunk 8]
```diff
@@ -44,6 +44,26 @@ <h2>${Advanced}</h2>
                         </div>
                         <div class="fieldDescription">${LabelCachePathHelp}</div>
                     </div>
+
+                    <div class="inputContainer">
+                        <div style="display: flex; align-items: center;">
+                            <div style="flex-grow:1;">
+                                <input is="emby-input" id="txtMetadataPath" label="${LabelMetadataPath}" autocomplete="off" />
```
Comment (JustAMan) [inline src/dashboardgeneral.html]: why `autocomplete=off`?

File: src/dashboardgeneral.html [hunk 9]
```diff
@@ -44,6 +44,26 @@ <h2>${Advanced}</h2>
...
+                            </div>
+                            <button type="button" is="paper-icon-button-light" id="btnSelectMetadataPath" title="${ButtonSelectDirectory}" class="emby-input-iconbutton"><i class="md-icon">search</i></button>
+                        </div>
+                        <div class="fieldDescription">${LabelMetadataPathHelp}</div>
+                    </div>
+
+                    <div class="inputContainer">
+                        <div style="display: flex; align-items: center;">
+                            <div style="flex-grow:1;">
+                                <input is="emby-input" id="txtMetadataNetworkPath" label="${LabelOptionalNetworkPath}" autocomplete="off" />
```
Comment (JustAMan) [inline src/dashboardgeneral.html]: again, why autocomplete off?

File: src/dlnasettings.html [hunk 10]
```diff
@@ -10,11 +10,6 @@
                         <h2 class="sectionTitle">${TabSettings}</h2>
                         <a is="emby-linkbutton" class="raised button-alt headerHelpButton" target="_blank" href="https://web.archive.org/web/20181216120305/https://github.com/MediaBrowser/Wiki/wiki/Dlna-settings">${Help}</a>
                     </div>
-
```
Comment (JustAMan) [inline src/dlnasettings.html]: hmm, why remove?

File: src/scripts/notificationsettings.js [hunk 11]
```diff
@@ -2,29 +2,55 @@ define(["loading", "libraryMenu", "globalize", "listViewStyle", "emby-linkbutton
...
-        loading.show(), ApiClient.getJSON(ApiClient.getUrl("Notifications/Types")).then(function(list) {
-            var html = "",
-                lastCategory = "",
-                showHelp = !0;
+        loading.show();
+        ApiClient.getJSON(ApiClient.getUrl("Notifications/Types")).then(function(list) {
+            var html = "";
+            var lastCategory = "";
+            var showHelp = true;
             html += list.map(function(i) {
```
Comment (JustAMan) [inline src/scripts/notificationsettings.js]: `i` -> `notificationType` ;)

File: src/scripts/notificationsettings.js [hunk 12]
```diff
@@ -2,29 +2,55 @@ define(["loading", "libraryMenu", "globalize", "listViewStyle", "emby-linkbutton
...
+                        showHelp = false;
+                        itemHtml += '<a is="emby-linkbutton" class="raised button-alt headerHelpButton" target="_blank" href="https://web.archive.org/web/20181216120305/https://github.com/MediaBrowser/Wiki/wiki/Notifications">';
+                        itemHtml += globalize.translate("Help");
+                        itemHtml += "</a>";
+                    }
+                    itemHtml += "</div>";
+                    itemHtml += '<div class="paperList">';
+                }
+                itemHtml += '<a class="listItem listItem-border" is="emby-linkbutton" data-ripple="false" href="notificationsetting.html?type=' + i.Type + '">';
+                i.Enabled ? itemHtml += '<i class="listItemIcon md-icon">notifications_active</i>' : itemHtml += '<i class="listItemIcon md-icon" style="background-color:#999;">notifications_off</i>';
```
Comment (JustAMan) [inline src/scripts/notificationsettings.js]:
```suggestion
                if (i.Enabled) {
                    itemHtml += '<i class="listItemIcon md-icon">notifications_active</i>';
                } else {
                    itemHtml += '<i class="listItemIcon md-icon" style="background-color:#999;">notifications_off</i>';
                }
```

File: src/scripts/plugincatalogpage.js [hunk 13]
```diff
@@ -41,82 +41,31 @@ define(["loading", "libraryMenu", "globalize", "cardStyle", "emby-linkbutton", "
...
-                html += getPluginHtml(topPlugins[i__q], options, installedPlugins);
-            }
-
-            html += "</div>";
-            html += "</div>";
-        }
-
         var hasOpenTag = false;
-
         for (currentCategory = null, false === options.showCategory && (html += '<div class="itemsContainer vertical-wrap">', hasOpenTag = true), i__q = 0, length = availablePlugins.length; i__q < length; i__q++) {
```
Comment (JustAMan) [inline src/scripts/plugincatalogpage.js]: care to de-uglify this for loop?

File: src/scripts/plugincatalogpage.js [hunk 14]
```diff
@@ -41,82 +41,31 @@ define(["loading", "libraryMenu", "globalize", "cardStyle", "emby-linkbutton", "
...
-
-            if (bName > aName) {
+            } else if (b.name > a.name) {
                 return -1;
             }
-
             return 0;
         });
+
         var i__q;
```
Comment (JustAMan) [inline src/scripts/plugincatalogpage.js]: pls give this variable a more descriptive name... or if it is really an index variable rename it to just `i`

File: src/dashboard/librarysettings.js (hunk 2 above)
Comment (dkanada) [reply src/dashboard/librarysettings.js]: It was one of the dead settings, there was no UI for this on the page.

File: src/dashboard/librarysettings.js (hunk 3 above)
Comment (dkanada) [reply src/dashboard/librarysettings.js]: None of the three had UI elements, they must have been old settings.

File: src/dashboard/livetvtuner.js (hunk 7 above)
Comment (dkanada) [reply src/dashboard/livetvtuner.js]: The only two tabs were the main page and the services page, which I removed.

File: src/dashboardgeneral.html (hunk 8 above)
Comment (dkanada) [reply src/dashboardgeneral.html]: I was just leaving that in from the other examples, didn't see a reason to change it specifically. Do you think it should be enabled?

File: src/dlnasettings.html (hunk 10 above)
Comment (dkanada) [reply src/dlnasettings.html]: so more code can fit on my screen :p there are a lot of newlines

Comment (JustAMan) [reply src/dlnasettings.html]: I was referring to the whole removed block, it just happened to start with newline

1.

Comment (dkanada) [reply src/dlnasettings.html]: Ah that one was in the description.

> The DLNA page had dead code referencing the streaming bitrate limit, which is currently in the playback section.

File: src/dashboard/dashboardgeneral.js (hunk 1 above)
Comment (dkanada) [reply src/dashboard/dashboardgeneral.js]: I would rather leave that alone for this batch of changes, I am just moving things around for now.

**Comment by cvium** [comment]
Network share path is _per_ path and not a global setting...

File: src/dashboardgeneral.html (hunk 8 above)
Comment (JustAMan) [reply src/dashboardgeneral.html]: I actually have no preference neither for `on` nor `off` TBH :) So was just curious...

**Comment by dkanada** [comment]
This pull request is now waiting on the [plugin](https://github.com/dkanada/jellyfin-plugin-opensubtitles) to get moved to the organization.
